src/structural_properties.c

```diff
@@ -6006,36 +6006,50 @@ int igraph_i_avg_nearest_neighbor_degree_weighted(const igraph_t *graph,
  * Average nearest neighbor degree.
  *
  * Calculates the average degree of the neighbors for each vertex, and
- * optionally, the same quantity in the function of vertex degree.
+ * optionally, the same quantity as a function of the vertex degree.
  *
- * </para><para>For isolate vertices \p knn is set to
- * NaN. The same is done in \p knnk for vertex degrees that
+ * </para><para>
+ * For isolated vertices \p knn is set to NaN.
+ * The same is done in \p knnk for vertex degrees that
  * don't appear in the graph.
  *
- * \param graph The input graph, it can be directed but the
- *   directedness of the edges is ignored.
+ * </para><para>
+ * The weighted version computes a weighted average of the neighbor degrees as
+ *
+ * <code>k_nn_u = 1/s_u sum_u w_uv k_v</code>,
+ *
+ * where <code>s_u</code> is the sum of the incident edge weights of vertex \c u,
+ * i.e. its strength. The sum runs over the neighbors \c v of vertex \c u
```

Comment:
I would write here, 
`edge weights of vertex \c u, i.e. its strenght, sum_u w_uv`.